src/providers/date-time/date-time.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+
+import { DateTime } from '../../shared/helpers/date-time';
+import { AppConfigProvider } from '../app-config/app-config';
+import { isEmpty } from 'lodash';
+
+@Injectable()
+export class DateTimeProvider {
+
+  constructor(private appConfigProvider: AppConfigProvider) {}
+
+  public now(): DateTime {
```

Comment:
Should we also remove the `.now()` method from the date-time helper so we are forced to use this?

Reply:
I see what you mean, I was thinking about that, but the constructor would still be available as public and we need to instantiate DateTime somehow.

What do you think about making the constructor to have required parameter, which would be a `DateTime`, `string` or `Date`?

Reply:
could have an optional parameter on the constructor, and default to current date time if not supplied.

Reply:
I am removing `DateTime.now()` which will encourage us to use the `DateTimeProvider.now()` instead.